stablehlo/dialect/TypeInference.cpp

```diff
@@ -3614,6 +3614,43 @@ LogicalResult verifyDotOp(std::optional<Location> location,
   return success();
 }
 
+LogicalResult verifyDotGeneralOpQuantizationConstraints(
+    std::optional<Location> location, Type lhsType, Type rhsType,
+    Type resultType, ArrayRef<int64_t> rhsContractingDimensions) {
+  Type lhsElementType = getElementTypeOrSelf(lhsType);
+  Type rhsElementType = getElementTypeOrSelf(rhsType);
+  Type resultElementType = getElementTypeOrSelf(resultType);
+
+  // dot_general_c15
+  if (auto rhsPerTensorQuantType =
+          dyn_cast<quant::UniformQuantizedType>(rhsElementType)) {
+    if (rhsPerTensorQuantType.getZeroPoint() != 0) {
+      return emitOptionalError(location,
+                               "Zero point of rhs of dot_general should be 0");
+    }
+  } else if (auto rhsPerAxisQuantType =
+                 dyn_cast<quant::UniformQuantizedPerAxisType>(rhsElementType)) {
+    if (llvm::any_of(rhsPerAxisQuantType.getZeroPoints(),
+                     [](int64_t zero_point) { return zero_point != 0; })) {
+      return emitOptionalError(location,
+                               "Zero points of rhs of dot_general should be 0");
+    }
+
+    // dot_general_c16
+    if (llvm::find(rhsContractingDimensions,
+                   rhsPerAxisQuantType.getQuantizedDimension()) !=
+        rhsContractingDimensions.end()) {
+      return emitOptionalError(location,
+                               "Quantization dimension should not be a "
```

Comment:
Quantization dimension of rhs should not be in the contracting... ?

Reply:
Done.